sos/report/plugins/snap_service.py

```diff
@@ -0,0 +1,86 @@
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.report.plugins import Plugin, IndependentPlugin
+
+
+class SnapRpc(Plugin, IndependentPlugin):
+    """
+    Collects snap service information and logs.
+    This plugin captures:
+    - snap service logs
+    - snap service configuration
+    - runtime snap service information
+    - runtime spdk information
+    """
+
+    short_desc = "NVIDIA snap service"
+    plugin_name = "snap_service"
+
+    def check_enabled(self):
+        return self._get_snap_container_id() is not None
+
+    def setup(self):
+        self.collect_snap_rpc_logs()
+        self.collect_snap_config()
+        commands = self.get_running_snap_rpc_commands()
+        if commands:
+            SPDK_RPC_CMD, SNAP_RPC_CMD = commands
+            self.collect_spdk_rpc(SPDK_RPC_CMD)
+            self.collect_snap_rpc(SNAP_RPC_CMD)
+
+    def get_running_snap_rpc_commands(self):
+        container_id = self._get_snap_container_id()
+        if not container_id:
+            return None
+        CONTAINER_CMD = f"crictl exec {container_id}"
+        SPDK_RPC_CMD = f"{CONTAINER_CMD} spdk_rpc.py"
+        SNAP_RPC_CMD = f"{CONTAINER_CMD} snap_rpc.py"
+        return SPDK_RPC_CMD, SNAP_RPC_CMD
+
+    def collect_snap_rpc_logs(self):
+        self.add_copy_spec(
+            [
+                "/var/log/snap-log/rpc-log",
+            ]
+        )
+
+    def collect_snap_config(self):
+        self.add_copy_spec(
+            [
+                "/opt/nvidia/nvda_snap/bin/spdk_config.json",
+                "/opt/nvidia/nvda_snap/bin/snap_rpc_init.conf",
+                "/opt/nvidia/nvda_snap/bin/set_environment_variables.sh",
+            ]
+        )
+
+    def collect_spdk_rpc(self, spdk_rpc_cmd):
+        self.add_cmd_output(
+            f"{spdk_rpc_cmd} bdev_get_bdevs",
+            timeout=30,
+        )
+
+    def collect_snap_rpc(self, snap_rpc_cmd):
+        self.add_cmd_output(
+            [
+                f"{snap_rpc_cmd} emulation_resource_list",
+                f"{snap_rpc_cmd} emulation_function_list",
+                f"{snap_rpc_cmd} nvme_subsystem_list",
+                f"{snap_rpc_cmd} snap_global_param_list",
+                f"{snap_rpc_cmd} virtio_blk_controller_list",
+            ],
+            timeout=30,
+        )
+
+    def _get_snap_container_id(self):
+        snap_container_cmd = "crictl ps -s running -q --name snap"
```

Comment:
crictl may not be installed, you can add dependency for that similar to https://github.com/NVIDIA/doca-sosreport/blob/83dc5855baba86cd9e8a49e8f4c0f8dc62ca0caa/sos/report/plugins/crio.py#L20-L21